src/interaction/FixedTripleListTypesInteractionTemplate.hpp

```diff
@@ -0,0 +1,270 @@
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// ESPP_CLASS
+#ifndef _INTERACTION_FIXEDTRIPLELISTTYPESINTERACTIONTEMPLATE_HPP
+#define _INTERACTION_FIXEDTRIPLELISTTYPESINTERACTIONTEMPLATE_HPP
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+
+#include "types.hpp"
+#include "Interaction.hpp"
+#include "Real3D.hpp"
+#include "Tensor.hpp"
+#include "Particle.hpp"
+#include "FixedTripleList.hpp"
+#include "esutil/Array3D.hpp"
+#include "bc/BC.hpp"
+#include "SystemAccess.hpp"
+
+#include "storage/Storage.hpp"
+
+namespace espressopp {
+namespace interaction {
+template<typename _Potential>
+class FixedTripleListTypesInteractionTemplate : public Interaction, SystemAccess {
+ protected:
+  typedef _Potential Potential;
+
+ public:
+  FixedTripleListTypesInteractionTemplate
+      (shared_ptr<System> system,
+       shared_ptr<FixedTripleList> _fixedtripleList)
+      : SystemAccess(system), fixedtripleList(_fixedtripleList) {
+    potentialArray = esutil::Array3D<Potential, esutil::enlarge>(0, 0, 0, Potential());
+    ntypes = 0;
+  }
+
+  virtual ~FixedTripleListTypesInteractionTemplate() { }
+
+  void setFixedTripleList(shared_ptr<FixedTripleList> _fixedtripleList) {
+    fixedtripleList = _fixedtripleList;
+  }
+
+  shared_ptr<FixedTripleList> getFixedTripleList() {
+    return fixedtripleList;
+  }
+
+  void setPotential(int type1, int type2, int type3, const Potential &potential) {
+    // typeX+1 because i<ntypes
+    ntypes = std::max(ntypes, std::max(std::max(type1 + 1, type2 + 1), type3 + 1));
+    potentialArray.at(type1, type2, type3) = potential;
+    if (type1 != type3) {  // add potential in the other direction
+      potentialArray.at(type3, type2, type1) = potential;
+    }
+  }
+
+  // this is used in the innermost force-loop
+  Potential &getPotential(int type1, int type2, int type3) {
+    return potentialArray.at(type1, type2, type3);
+  }
+
+  shared_ptr<Potential> getPotentialPtr(int type1, int type2, int type3) {
+    return make_shared<Potential>(potentialArray.at(type1, type2, type3));
+  }
+
+  virtual void addForces();
+  virtual real computeEnergy();
+  virtual real computeEnergyAA();
+  virtual real computeEnergyCG();
+  virtual void computeVirialX(std::vector<real> &p_xx_total, int bins);
+  virtual real computeVirial();
+  virtual void computeVirialTensor(Tensor &w);
+  virtual void computeVirialTensor(Tensor &w, real z);
+  virtual void computeVirialTensor(Tensor *w, int n);
+  virtual real getMaxCutoff();
+  virtual int bondType() { return Pair; }
```

Comment:
Isn't this 'return Angular' ?

(The logger in this interaction template is also still referring to FixedPairList instead of FixedTripleList)